ui/components/app/nft-details/nft-details.js

```diff
@@ -94,10 +97,26 @@ export default function NftDetails({ nft }) {
     'M/d/y',
   );
 
-  const onRemove = () => {
-    dispatch(removeAndIgnoreNft(address, tokenId));
-    dispatch(setNewNftAddedMessage(''));
-    dispatch(setRemoveNftMessage('success'));
+  const onRemove = async () => {
+    try {
+      await dispatch(removeAndIgnoreNft(address, tokenId));
+      dispatch(setNewNftAddedMessage(''));
+      dispatch(setRemoveNftMessage('success'));
+    } catch (err) {
+      dispatch(setNewNftAddedMessage(''));
+      dispatch(setRemoveNftMessage('error'));
+    }
+    // track event
+    trackEvent({
```

Comment:
It looks like the trackEvent function is called after the try-catch block, which means it will execute regardless of whether the `removeAndIgnoreNft` operation succeeds or fails. 

If the intention is to only track successful NFT removal events, I recommend moving the trackEvent call inside the try block, right after the `dispatch(setRemoveNftMessage('success'))`; line. 

This way, we ensure that events are only tracked when the operation completes successfully, providing us with more accurate analytics data.

Reply:
Initially I ve put the trackEvent function only when successful. Then i thought that even if for some reason the request to background fails, we would still wanna catch that the user wanted to remove that NFT

Reply:
Yup sounds the property would be useful, i can add that 🙏